src/Utils/Proxy/HttpProxyAgent.ts

```diff
@@ -0,0 +1,33 @@
+import { Wildcard } from '../Wildcard';
+import {
+  HttpProxyAgent as ProxyAgent,
+  type HttpProxyAgentOptions as AgentOptions
+} from 'http-proxy-agent';
+import { URL } from 'url';
+import http, { globalAgent } from 'http';
+
+export type HttpProxyAgentOptions<T> = AgentOptions<T> & {
+  domains?: string[];
+};
+
+export class HttpProxyAgent<T extends string> extends ProxyAgent<T> {
+  private domains: Wildcard;
+
+  constructor(proxy: T | URL, opts: HttpProxyAgentOptions<T>) {
+    super(proxy, opts);
+
+    if (opts.domains) {
+      this.domains = new Wildcard(opts.domains ?? []);
+    }
+  }
+
+  public addRequest(req: http.ClientRequest, opts: http.RequestOptions): void {
+    if (this.domains && this.domains.match(opts.hostname)) {
+      // @ts-expect-error `addRequest()` isn't defined in `@types/node`
+      super.addRequest(req, opts);
+    } else {
+      // @ts-expect-error `addRequest()` isn't defined in `@types/node`
```

Comment:
the same thing was  done in `http-proxy-agent` library https://github.com/TooTallNate/proxy-agents/blob/main/packages/http-proxy-agent/src/index.ts#L76C1-L77C1

would gladly follow any recommendations to improve this.

Reply:
the other solution is to not override proxy agents but do application level handling of proxy domains.
So, you can adjust proxy factory that is used in rest endpoints(like RunScan command) to check proxyDomains with relation to cluster/hostname. For HttpRequestExecutor you can check `request.url` when passing `Agent` to `http.request`
Overriding proxy agents does not look safe as it seems there still be connection establishment with the proxy server (see https://github.com/TooTallNate/proxy-agents/blob/5555794b6d9e4b0a36fac80a2d3acea876a8f7dc/packages/http-proxy-agent/src/index.ts#L125) that could lead to any sorts of undesired behaviors:
what if connect fails? that for sure can happen as they could refuse proxying provided host (see https://github.com/TooTallNate/proxy-agents/blob/5555794b6d9e4b0a36fac80a2d3acea876a8f7dc/packages/https-proxy-agent/src/index.ts#L111)

Reply:
Could you elaborate on this a little please ?

You mean to create two clients (both proxy and non-proxy) in RestScan and when sending a request check with proxyDomains list and choose which one to use ?

Reply:
But we're not overriding the connect method, right ? Only addRequest here.

Reply:
> You mean to create two clients (both proxy and non-proxy) in RestScan and when sending a request check with proxyDomains list and choose which client to use

Basically yes but two clients are not necessary as during lifecycle of rest stuff `hostname` CLI option never changes(you don't need full url for proxy domains check in rest api scenarios) and you can return `{}` from `ProxyFactory.createProxy`.

> But we're not overriding the connect method, right ? Only addRequest here.

I don't see you're overriding `connect` meaning that it will still connect to proxy when executing request to endpoints excluded from proxy domains

Reply:
Agree, but there can be repeater/bridge requests which may vary upon execution of scans. Like run scan on abc.com, then sub.abc.com, etc. Clients may want to proxy only sub.abc.com only.

Reply:
@maksadbek for repeater requests there `HttpRequestExecutor` with its own "factory" where you can adjust the behavior:
https://github.com/NeuraLegion/bright-cli/blob/e1570e869c9e1391d986ef56e018ce4b27df303e/src/RequestExecutor/HttpRequestExecutor.ts#L192

Reply:
Gotcha, sure that makes sense. I'll amend the PR

Reply:
@denis-maiorov-brightsec I've amended the PR and removed the proxy part for Bright endpoints, as it's an obsolete and in this change we want ONLY selectively proxy requests to client's servers. Requests to Bright servers won't be proxied. 

So I've added only changes to HTTPRequestExecutor.ts